include/ipr/interface

```diff
@@ -124,19 +124,23 @@ namespace ipr {
    struct Array_delete;          // array delete-expression     delete[] p
    struct Complement;            // bitwise complement                  ~m
    struct Delete;                // delete-expression             delete p
+   struct Demote;                // integral or floating-point conversion
    struct Deref;                 // dereference expression              *p
    struct Expr_list;             // expression list        
    struct Sizeof;                // sizeof expression: sizeof *p or sizeof(int)
    struct Typeid;                // typeid expression, like sizeof expression
    struct Id_expr;               // an identifier used in an expression
    struct Label;                 // a label - target of a goto-statement
                                  //         or entry of a switch-statement
+   struct Materialize;           // temporary materialization        T().m
    struct Not;                   // logical negation                 !cond
    struct Paren_expr;            // parenthesized expression           (a)
    struct Post_decrement;        // post-increment                     p++
    struct Post_increment;        // post-decrement                     p--
    struct Pre_decrement;         // pre-increment                      ++p
    struct Pre_increment;         // pre-decrement                      --p
+   struct Promote;               // integral or floating-point promotion
+   struct Read;                  // lvalue to rvalue conversion      = var
```

Comment:
The syntax `= var` is a bit misleading here: assignment of class types don't generally require lvalue-to-rvalue conversion.  Let's remove it